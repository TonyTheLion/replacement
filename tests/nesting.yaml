---
# nesting blocks
# examples/nesting.yaml
replacement:
  # insert '1.1' into metadata (substitutions dict) under the key 'version'
  - meta: text
    spec: version
    input: 1.1
  - text: text
    proc: format
    input: |
      version {version}
  - text: text
    input:
      # metadata additions/changes seen by later blocks and any children,
      # but seen outside this list
      - meta: text
        spec: version
        input: 1.0
      - text: text
        proc: format
        input: |
          version {version} is clobbered in inner scope
      - text: file
        input:
          hello.out
  - text: text
    proc: format
    input: |
      outer version is still {version}
...
