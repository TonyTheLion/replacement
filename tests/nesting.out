version 1.1
version 1.0 is clobbered in inner scope
hello world
outer version is still 1.1
